system/Entity.php

```diff
@@ -421,6 +421,8 @@ protected function mutateDate($value)
 
 	protected function castAs($value, string $type)
 	{
+		if(substr($type,0,1) === '?' && $value === null) return null;
+
 		switch($type)
 		{
 			case 'integer':
```

Comment:
@nowackipawel Since you're around here, could you also add `int` as a case here and `bool` with the boolean below? Thanks.

Reply:
I think we shouldn't use aliases but I added it with additional comment that bool/int are aliases.

Reply:
The more correct return types are int and bool and not integer and boolean. https://github.com/codeigniter4/coding-standard/issues/22 . Currently,  integer and boolean are the aliases.

Test:

```php
<?php

function foo(?integer $x):int {
	return 2 * $x;
}

echo foo(5);

```

Reply:
So, the comment is wrong.

Reply:
You know... or not I'm lazy and I have almost 60 entities where I use integer and boolean as was in docs :) ... and I don't like aliases and I don't want to change classes.... but ofc you are right tl;dr; I've changed it.